homeassistant/components/vallox/sensor.py

```diff
@@ -127,6 +130,27 @@ def native_value(self) -> StateType:
         return VALLOX_CELL_STATE_TO_STR.get(super_native_value)
 
 
+class ValloxProfileDurationSensor(ValloxSensorEntity):
+    """Child class for profile duration reporting."""
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor."""
+        minutes_left: int | float | None = 0
+
+        if Profile.EXTRA is self.coordinator.data.profile:
+            minutes_left = self.coordinator.data.get("A_CYC_EXTRA_TIMER", 0)
+        elif Profile.FIREPLACE is self.coordinator.data.profile:
+            minutes_left = self.coordinator.data.get("A_CYC_FIREPLACE_TIMER", 0)
+        elif Profile.BOOST is self.coordinator.data.profile:
+            minutes_left = self.coordinator.data.get("A_CYC_BOOST_TIMER", 0)
+
+        if not isinstance(minutes_left, int):
+            return 0
```

Comment:
Is 0 a good indicator for this case? Is it allowed/ok to use something like "unknown"? 🤔 Or even "unavailable"?

Reply:
I just used that as a defensive mechanism for the typing, it's technically always an int but the `MetricValue` type is an union of `int | float | None`

not sure what's a pythonesque way to do it, in typescript i'd just do `...get(...) as T`

Reply:
so `None` for "unexpected" value AND default value? so that the value is `None` when none of the timed profiles are on?

Reply:
Yeah, I think that could make sense. Maybe a specific early return for the normal profile, and then we could even log something on debug for unexpected values, or WDYT?

Reply:
In any case, I really like this new sensor 👍 The thought that the Vallox unit could provide it never crossed my mind, even though I almost implemented something like it using template sensors in Home Assistant at some point 😀